pnpbridge/src/pnpbridge/inc/pnpadapter_api.h

```diff
@@ -168,32 +184,79 @@ extern "C"
     );
 
     /**
-    * @brief    PnpInterfaceHandleSetContext sets context on an interface handle
+    * @brief    PnpComponentHandleSetContext sets device context on a component handle
 
-    * @param    InterfaceHandle          Handle to pnp adapter
+    * @param    ComponentHandle          Handle to pnp component
     *
-    * @param    InterfaceContext         Interface context to set
+    * @param    ComponentDeviceContext   Component's device context to set
     *
-    * @returns  void  PnpInterfaceHandleSetContext will always overwrite the interface context
+    * @returns  void  PnpComponentHandleSetContext will always overwrite the device context
     */
     MOCKABLE_FUNCTION(,
         void,
-        PnpInterfaceHandleSetContext,
-        PNPBRIDGE_INTERFACE_HANDLE, InterfaceHandle,
-        void*, InterfaceContext
+        PnpComponentHandleSetContext,
+        PNPBRIDGE_COMPONENT_HANDLE, ComponentHandle,
+        void*, ComponentDeviceContext
     );
 
     /**
-    * @brief    PnpInterfaceHandleGetContext gets adapter context from an adapter handle
+    * @brief    PnpComponentHandleGetContext gets device context from a component handle
 
-    * @param    AdapterHandle          Handle to pnp interface
+    * @param    ComponentHandle        Handle to pnp component
     *
-    * @returns  void *                 Interface context
+    * @returns  void *                 Component's device context
     */
     MOCKABLE_FUNCTION(,
         void*,
-        PnpInterfaceHandleGetContext,
-        PNPBRIDGE_INTERFACE_HANDLE, InterfaceHandle
+        PnpComponentHandleGetContext,
+        PNPBRIDGE_COMPONENT_HANDLE, ComponentHandle
+    );
+
+    /**
+    * @brief    PnpComponentHandleSetPropertyUpdateCallback sets property update callback on the
+    *           component handle to manage cloud to device property updates per component
+
+    * @param    ComponentHandle        Handle to pnp component
+    *
+    * @param    PropertyUpdateCallback Property update Callback
+    * 
+    * @returns  void   PnpComponentHandleSetPropertyUpdateCallback will always overwrite successfully
+    */
+    MOCKABLE_FUNCTION(,
+        void,
+        PnpComponentHandleSetPropertyUpdateCallback,
+        PNPBRIDGE_COMPONENT_HANDLE, ComponentHandle,
+        PNPBRIDGE_COMPONENT_PROPERTY_CALLBACK, PropertyUpdateCallback
+    );
+
+    /**
+    * @brief    PnpComponentHandleSetCommandCallback sets process command callback on the
+    *           component handle to manage cloud to device command updates per component
+
+    * @param    ComponentHandle        Handle to pnp component
+    *
+    * @param    CommandCallback        Process command callback
+    * 
+    * @returns  void   PnpComponentHandleSetCommandCallback will always overwrite successfully
+    */
+    MOCKABLE_FUNCTION(,
+        void,
+        PnpComponentHandleSetCommandCallback,
```

Comment:
>PnpComponentHandleSetCommandCallback [](start = 8, length = 36)

Please add "PnpComponentHandleSetCommandCallback should be called from a PNPBRIDGE_COMPONENT_CREATE callback"